[docs-only] Add documentation for proxy routes (#7771)
owncloud/ocis: services/proxy/README.md

-   Signed URL
-   Public Share Token

## Configuring routes

The proxy handles routing to all endpoints that ocis offers. The current default routes can be found here https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go

> Due to complexity of the configuration, new routes can only be defined in the yaml file.
> For overwriting the default routes use:

Reviewer: ```suggestion
Due to the complexity when defining routes, these can only be defined in the yaml file but not via environment variables.

For _overwriting_ default routes, use the following yaml example:

```

```yaml
policies:
  - name: ocis
    routes:
      - endpoint: /
        service: com.owncloud.web.web
      - endpoint: /dav/
        service: com.owncloud.web.ocdav
```
For adding additional routes to the default routes use:
```yaml
additional_policies:
  - name: ocis
    routes:
      - endpoint: /custom/endpoint
        service: com.owncloud.cutom.custom
```

> A route has the following configurable parameters:

Reviewer: ```suggestion

A route has the following configurable parameters:

```

```yaml

> endpoint: "" # the url that should be routed
> service: "" # the service the url should be routed to
> unprotected: false # skip authorization

Reviewer: ```suggestion
endpoint: ""       # the url that should be routed
service: ""        # the service the url should be routed to
unprotected: false # skip authorization
```
Renders hashes underneath making the keys better readble
What happens with `unprotected` true?

Author: `unprotected: true` => No authorization checks. This is an open endpoint
`unprotected: false` => Authorization is checked on request, user needs an auth header to access this endpoint

-   Signed URL
-   Public Share Token

## Configuring Routes

> The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go).

Reviewer: ```suggestion
The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary, when writing own ocis extensions.
```

> The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary, when writing own ocis extensions.

Reviewer: ```suggestion
The proxy handles routing to all endpoints that ocis offers. The currently availabe default routes can be found [in the code](https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go). Changing or adding routes can be necessary when writing own ocis extensions.
```

Due to the complexity when defining routes, these can only be defined in the yaml file but not via environment variables.

For _overwriting_ default routes, use the following yaml example:

```yaml
policies:
  - name: ocis
    routes:
      - endpoint: /
        service: com.owncloud.web.web
      - endpoint: /dav/
        service: com.owncloud.web.ocdav
```

For adding _additional_ routes to the default routes use:

```yaml
additional_policies:
  - name: ocis
    routes:
      - endpoint: /custom/endpoint

>         service: com.owncloud.cutom.custom

Reviewer: ```suggestion
        service: com.owncloud.custom.custom
```
typo ? 🤣
